packages/odyssey-react-mui/src/Tag.tsx

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { Chip } from ".";
+import { memo, useContext } from "react";
+import { TagListContext } from "./TagListContext";
+
+export type TagProps = {
+  isDisabled?: boolean;
+  label: string;
+  onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+  onDelete?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+};
```

Comment:
```suggestion
  onClick?: ClickEventHandler<HTMLDivElement>;
  onDelete?: ClickEventHandler<HTMLDivElement>;
};
```

`ClickEventHandler` comes from the "react" library.

If this doesn't work, another safe way to do callbacks is referencing the MUI type directly:
```suggestion
  onClick?: MuiTagProps["onClick"];
  onDelete?: MuiTagProps["onDelete"];
};
```

I actually prefer this because we don't control these props, and MUI knows it's a `div`. We can't guarantee it'll always be a `<div>` that from our end.

Even if I don't like their types, this is the safest way for functions like this. Ideally, we would completely control it, but we can't always do that especially when it's relative to the underlying HTML.